core/src/main/java/org/infinispan/container/offheap/MemoryAddressHash.java

```diff
@@ -19,32 +20,32 @@
    public MemoryAddressHash(int pointers) {
       this.pointerCount = nextPowerOfTwo(pointers);
       long bytes = ((long) pointerCount) << 3;
-      memory = UNSAFE.allocateMemory(bytes);
+      memory = OffHeapMemory.allocate(bytes);
       // Have to clear out bytes to make sure no bad stuff was read in
       UNSAFE.setMemory(memory, bytes, (byte) 0);
    }
 
-   private int findOffset(Object instance) {
+   private long findOffset(Object instance) {
```

Comment:
We need a stress test with > 500m entries ;)

Reply:
Have a machine to "lend" me :)